src/main/cpp/t07_ip.cpp

```diff
@@ -29,10 +29,50 @@
 
 #include "t07_ip.h"
 #include <iostream>
-
+#include <string>
 
 using namespace std;
 
 int t07_ip() {
+	string h, number = "";
+	bool pass = true, num = true;
+	int numofpoints = 0;
+	int numofnumbers = 0;
+	cin >> h;
+	for (int i = 0; i < h.size(); i++)
+	{
+		if (h[i] == '.' && num ==false)
+		{
+			if (stoi(number) > 255)
+			{
 
+				pass = false;
+				break;
+			}
+			if (!num)
+			{
+				num = true;
+				numofpoints++;
+			}
+			number = "";
+		}
+		else if (((int)h[i] < 48 || (int)h[i] > 57) || ((h[i] == h[i + 1]) && h[i] == '.'))
+		{
+			pass = false;
+			break;
+		}
+		else if ((int)h[i] >= 48 || (int)h[i] <= 57)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Logical disjunction always evaluates to true: EXPR >= 48 || EXPR <= 57.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4504172)